### PR TITLE
Bump rocblas to 3.8.0

### PR DESCRIPTION
**Pull request by haampie**
and deploy `rocm_smi` executable in `bin/` folder. This is used in Tensile, which is part of rocblas.

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocblas/package.py [hunk 1]
```diff
@@ -75,11 +81,16 @@ def cmake_args(self):
...
         ]
 
-        if '@3.7.0' in self.spec:
+        if '@3.7.0:' in self.spec:
             args.append('-DTensile_LIBRARY_FORMAT=msgpack')
 
+        # Potentially this already works in 3.7.0, but was not tested
+        if '@3.8.0:' in self.spec:
+            args.append('-DBUILD_WITH_TENSILE_HOST=ON')
+        else:
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocblas/package.py]: this is recommendation from the rocblas internal team -
ROCm 3.7 can use -DBUILD_WITH_TENSILE_HOST=ON while ROCm 3.8 and later must use -DBUILD_WITH_TENSILE_HOST=ON .  ROCm 3.5 must use -DBUILD_WITH_TENSILE_HOST=OFF .  Thanks.

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocblas/package.py]: i have tested the recipe for 3.8.0 and it works .

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocblas/package.py]: need fix for flake8 error

Comment (haampie) [reply var/spack/repos/builtin/packages/rocblas/package.py]: Thanks, it's done

**Review by srekolam — APPROVED** [review]
We are able to build

**Comment by haampie** [comment]
Perfect, ping @adamjstewart :)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**